### PR TITLE
Update free-programming-books-subjects.md

### PR DESCRIPTION
**Pull request by sahildhiman-nsutian**
## What does this PR do?

Add resource

### Description

A great eBook for beginners to start off with Dynamic Programming , be it for interviews or for coding competitions.

### Why is this valuable (or not)?

This book is really a gem since Dynamic Programming problems are always asked in every competitive coding competitions.
Moreover , in some cases only DP solution is acceptable .Thus , this book is of great help for DP concepts.

### How do we know it's really free?

This book is really free since it doesn't require any subscription. Neither does it require any email address for downloading. It is completely free to download.

### For book lists, is it a book? For course lists, is it a course? etc.

This is an eBook in English for Algorithms and Data Structures.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by sahildhiman-nsutian** [comment]
@ericallam @emkay @radar @Andrea   Hey everyone I have completed my first pull request .Kindly review it.

**Comment by davorpa** [comment]
Thanks for taking your time to fill PR description ❤

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -99,6 +99,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Data Structures and Algorithms: Annotated Reference with Examples](https://web.archive.org/web/20170715160229/http://dotnetslackers.com/Community/files/folders/data-structures-and-algorithms/entry30283.aspx) - G. Barnett and L. Del Tongo
 * [Data Structures Succinctly Part 1, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/datastructurespart1) (PDF, Kindle) (email address *requested*, not required)
 * [Data Structures Succinctly Part 2, Syncfusion](https://www.syncfusion.com/resources/techportal/ebooks/datastructurespart2) (PDF, Kindle) (email address *requested*, not required)
+* [Dynamic Programming for Coding Interviews: A Bottom-Up Approach to Problem](https://libgen.is/book/index.php?md5=CCAA5C6429A0954D69371D5B2B411060) (PDF)
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:102]: - Domain not allowed. Use author/publisher website
- Not free 🤔?? https://www.amazon.com/-/en/Meenakshi/dp/1946556696
- Missing author attributions

**Comment by davorpa** [comment]
### :information_source: INVALID

We appreciate your desire to want to contribute but this appears not to be an authorized distribution. Please see [CONTRIBUTING.md](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md) guidelines.

> - we don't accept files hosted on google drive, dropbox, mega, scribd, issuu and other similar file upload platforms
> - use the link with the most authoritative source (meaning the author's website is better than the editor's website, which is better than a third party website)

**Pull request closed without merging**